sources/api_client.go

```diff
@@ -282,3 +282,34 @@ func (sc *SourcesClient) createApplicationAuthentication(ctx context.Context, ap
 
 	return nil
 }
+
+// HealthCheck performs a lightweight health check against the Sources API
+// to verify connectivity and API availability
+func HealthCheck(ctx context.Context) error {
+	// Use a simple GET request to the openapi.json endpoint which is lightweight
+	reqURL, _ := url.Parse(fmt.Sprintf(
+		"%v://%v:%v/api/sources/v3.1/openapi.json", conf.SourcesScheme, conf.SourcesHost, conf.SourcesPort,
+	))
```

Comment:
May you please handle the error of this one too just in case? Just for completeness, because I don't think this will fail unless we provide a bad config...